pkg/helpers/generateClusterId.go

```diff
@@ -40,3 +45,30 @@ func ConvertCpuUsageToCores(millicpu float64) float64 {
 func ConvertMemoryUsageToMiB(bytes float64) float64 {
 	return bytes / 1024.0 / 1024.0
 }
+
+//error handling for decoding request body:
+func ErrorHandlingRequests(w http.ResponseWriter, err error) bool {
+	if err != nil {
+		var unmarshalTypeError *json.UnmarshalTypeError
+		var syntaxError *json.SyntaxError
+
+		switch {
+
+		case errors.As(err, &syntaxError):
+			http.Error(w, "{\"message\":\"Request body contains badly-formed JSON.\"}", http.StatusBadRequest)
+
+		case errors.As(err, &unmarshalTypeError):
+			msg := fmt.Sprintf("Request body contains an invalid value for the %s field", unmarshalTypeError.Field)
+			http.Error(w, msg, http.StatusBadRequest)
+
+		case errors.Is(err, io.EOF):
+			http.Error(w, "{\"message\":\"Request body must not be empty.\"}", http.StatusBadRequest)
+
+		default:
+			klog.Error(err.Error())
```

Comment:
Include some context in this error log.
```suggestion
			klog.Error("Unexpected error while processing request. ", err.Error())
```